ruby/define-method.md

```diff
@@ -0,0 +1,32 @@
+# Method Definition
+
+MRI Ruby provides syntax `def` that allow you to define a method.
+
+```ruby
+def my_method
+end
+
+# => :my_method
+```
+
+The returned value of a method definition is the symbol naming of the method. This is useful when you want to define and parse the method naming at the same time into other method. For example, you could expose helper method in Rails controller like this:
+
+```ruby
+def MyController < ActionController::Base
+  helper_method def my_helper
+                  # my_code
+                end
+end
+```
+
+Besides, it is also possible to define new instance method dynamically by taking advantage of metaprogramming techniques in Ruby with `define_method`. This method takes in a Proc, a Method or an UnboundMethod object. If a block is given, it will be used as the body of the method and will be evalulated with `instance_eval`. Example:
```

Comment:
Thoughts on removing the section on `define_method`? TILs should be concise, specific, and self-contained learnings. Instead of a tutorial on meta-programming methods in ruby, this could be just on how Ruby 2.1+ returns a symbol for method definition and how that can be used to inline helper method definitions.